design/speaker.md

```diff
@@ -0,0 +1,128 @@
+                   │                                       │ │               │
+┌───────────────┐  │ ┌───────────────┐  ┌───────────────┐  │ └───────────────┘
+│               │  │ │               │  │               │  │
+│   services    ├──┼─┤      k8s      ├──┤     main      ├──┤
+│  <<Service>>  │  │ │  <<package>>  │  │  <<package>>  │  │
+│               │  │ │               │  │               │  │
+└───────────────┘  │ └───────────────┘  └───────┬───────┘  │
+                   │                    <<use>> │          │ ┌───────────────┐
+                   │                            │          │ │               │
+┌───────────────┐  │reconcile           ┌───────▼───────┐  │ │    config     │
+│               │  │                    │               │  └─┤  <<package>>  │
+│     nodes     │  │                    │   Protocol    │    │               │
+│    <<Node>>   ├──┘                    │ <<interface>> │    └───────────────┘
+│               │                       │               │
+└───────────────┘                       └───────▲───────┘
+                                                │<<implements>>
+                                     ┌──────────┴────────────┐
+                                     │                       │
+                              ┌──────┴────────┐     ┌────────┴──────┐
+                              │     main::    │     │    main::     │
+                              │ L2_controller │     │ bgp_controller│
+                              │   <<class>>   │     │   <<class>>   │
+                              │               │     │               │
+                              └───────────────┘     └───────┬───────┘
+                                                            │
+                                                            │  <<use>>
+                                                    ┌───────▼───────┐
+                                                    │               │
+                                                    │    session    │
+                                                    │ <<interface>> │
+                                                    │               │
+                                                    └───────▲───────┘
+                                                            │ <<implements>>
+                                                            │
+                                                    ┌───────┴───────┐
+                                                    │               │
+                                                    │     bgp       │
+                                                    │  <<package>>  │
+                                                    │               │
+                                                    └───────────────┘
+
+```
+
+### main
+
+The main package contains the main() entrypoint for the speaker application. It
+creates an L2 and/or BGP controller, a speakerList and a k8s client. The k8s
```

Comment:
IMO we should use either "Layer 2" or "L2" consistently throughout the document.

Reply:
Changed to "Layer 2"